staging/src/k8s.io/apiserver/pkg/server/filters/cors_test.go

```diff
@@ -131,28 +133,31 @@ func TestCORSAllowedMethods(t *testing.T) {
 			http.HandlerFunc(func(http.ResponseWriter, *http.Request) {}),
 			[]string{".*"}, test.allowedMethods, nil, nil, "true",
 		)
-		server := httptest.NewServer(handler)
-		defer server.Close()
-		client := http.Client{}
-
-		request, err := http.NewRequest(test.method, server.URL+"/version", nil)
-		if err != nil {
-			t.Fatalf("unexpected error: %v", err)
-		}
-		request.Header.Set("Origin", "allowed.com")
+		var response *http.Response
+		func() {
+			server := httptest.NewServer(handler)
+			defer server.Close()
+			client := http.Client{}
+
+			request, err := http.NewRequest(test.method, server.URL+"/version", nil)
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			request.Header.Set("Origin", "allowed.com")
 
-		response, err := client.Do(request)
-		if err != nil {
-			t.Fatalf("unexpected error: %v", err)
-		}
+			response, err = client.Do(request)
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
 
-		methodAllowed := allowsMethod(response, test.method)
-		switch {
-		case test.allowed && !methodAllowed:
-			t.Errorf("Expected %v to be allowed, Got only %#v", test.method, response.Header.Get("Access-Control-Allow-Methods"))
-		case !test.allowed && methodAllowed:
-			t.Errorf("Unexpected allowed method %v, Expected only %#v", test.method, response.Header.Get("Access-Control-Allow-Methods"))
-		}
+			methodAllowed := allowsMethod(response, test.method)
```

Comment:
Move this block out of the func? We don't use the response var outside of the func now.

Reply:
make sense